mycroft/skills/intent_service.py

```diff
@@ -197,11 +219,11 @@ def remove_active_skill(self, skill_id):
         Arguments:
             skill_id (str): skill to remove
         """
-        for skill in self.active_skills:
+        for skill in copy(self.active_skills):
```

Comment:
if you cast this into a tuple instead of copying it, the intent of having an immutable copy for the loop would be more evident, IMO

Reply:
I disagree here (as per usual with me and Chris ;) ), an explicit copy is more expressive of the intention than an implicit copy as part of a cast.

Reply:
I love it. Before I started here I would take code review comments seriously and get upset on occasion but if this job has taught me anything it has taught me to be less thin skinned and more dismissive of stylistic input. FYI guys, my comment on this would have been "why is this not a lock? Casting and copy are not thread safe and have issues with the GIL"? Substance before style.

Reply:
I think the copy here is to handle the modification of the list below, not for thread safety, I'm sure a lock would be an improvement.